java/code/src/com/redhat/rhn/manager/contentmgmt/ContentManager.java

```diff
@@ -856,18 +850,39 @@ private static void alignErrata(Channel src, Channel tgt, Collection<ErrataFilte
         ErrataManager.mergeErrataToChannel(user, includedErrata, tgt, src, false, false);
     }
 
-    private static List<Long> extractPackageIds(Collection<Package> packages) {
-        return packages.stream().map(p -> p.getId()).collect(toList());
-    }
+    /**
+     * Filters given entities based on given filters.
+     *
+     * Entities are processed one-by-one by filters as follows:
+     * - when any DENY filter is satisfied for an entity -> this entity gets filtered out
+     * - when an ALLOW is satisfied for an entity -> this entity gets NOT filtered out (even if it had been filtered out
+     *   by a DENY filter = ALLOW filters have higher priority)
+     *
+     * @param entities entities, e.g. packages
+     * @param filters filters, e.g. package filters
+     * @param <T> the type of the entity (e.g. Package)
+     * @return Pair with a set of nonfiltered entities and a set filtered entities
```

Comment:
But what is what? We should document what of the Pair is included and excluded.

Reply:
Let me improve the comment.